client/boot/index.js

```diff
@@ -163,13 +163,17 @@ function boot() {
 
 		// Create layout instance with current user prop
 		Layout = require( 'layout' );
-		layout = React.render( React.createElement( Layout, {
```

Comment:
I must have squashed it into non-existence in #338, but in some of my earlier attempts to introduce a top-level provider, I used it as an opportunity to reduce some of the repetition here:

```diff
diff --git a/client/boot/index.js b/client/boot/index.js
index 99df3f4..ef7428d 100644
--- a/client/boot/index.js
+++ b/client/boot/index.js
@@ -163,13 +163,13 @@ function boot() {
 
                // Create layout instance with current user prop
                Layout = require( 'layout' );
-               layout = React.render( React.createElement( Layout, {
+               layout = React.createElement( Layout, {
                        user: user,
                        sites: sites,
                        focus: layoutFocus,
                        nuxWelcome: nuxWelcome,
                        translatorInvitation: translatorInvitation
-               } ), document.getElementById( 'wpcom' ) );
+               } );
        } else {
                analytics.setSuperProps( superProps );
 
@@ -179,12 +179,14 @@ function boot() {
                        LoggedOutLayout = require( 'layout/logged-out' );
                }
 
-               layout = React.render(
-                       React.createElement( LoggedOutLayout ),
-                       document.getElementById( 'wpcom' )
-               );
+               layout = React.createElement( LoggedOutLayout );
        }
 
+       layout = React.render(
+               React.createElement( ReduxProvider, { store: reduxStore }, () => layout ),
+               document.getElementById( 'wpcom' )
+       );
+
        debug( 'Main layout rendered.' );
 
        // If `?sb` or `?sp` are present on the path set the focus of layout
```